floodlight/io/dfl.py

```diff
@@ -0,0 +1,324 @@
+from floodlight.core.xy import XY
+
+
+def _read_pitch_from_mat_info(filepath_mat_info: Union[str, Path]) -> Pitch:
+    """Reads match_information XML file and returns the playing Pitch.
+
+    Parameters
+    ----------
+    filepath_mat_info: str or pathlib.Path
+        Full path to XML File where the Match Information data in DFL format is saved.
+
+    Returns
+    -------
+    pitch: Pitch
+        Pitch object with actual pitch length and width.
+    """
+    # set up XML tree
+    tree = etree.parse(str(filepath_mat_info))
+    root = tree.getroot()
+
+    # parse pitch
+    length = root.find("MatchInformation").find("Environment").get("PitchX")
+    length = float(length) if length else None
+    width = root.find("MatchInformation").find("Environment").get("PitchY")
+    width = float(width) if width else None
+    pitch = Pitch.from_template(
+        "dfl",
+        length=length,
+        width=width,
+    )
+
+    return pitch
+
+
+def _create_periods_from_dat(
+    filepath_dat: Union[str, Path]
+) -> Tuple[Dict[str, Tuple[int, int]], int]:
+    """Parses over position file and returns dictionary with periods
+
+    Parameters
+    ----------
+    filepath_dat: str or pathlib.Path
+        Path to XML File where the Position data in DFL format is saved
+
+    Returns
+    -------
+    periods: Dict[str, Tuple[int, int]
+        Dictionary with times for the segment:
+        `periods[segment] = (starttime, endtime)`
+    framerate: int
+        Temporal resolution of data in frames per second/Hertz.
+    """
+    periods = {}
+    framerate = None
+
+    # retrieve information from ball frame sets
+    for _, frame_set in etree.iterparse(filepath_dat, tag="FrameSet"):
+        if frame_set.get("TeamId") == "Ball":
+            frames = [frame for frame in frame_set.iterfind("Frame")]
+            periods[frame_set.get("GameSection")] = (
+                int(frames[0].get("N")),
+                int(frames[-1].get("N")),
+            )
+            if framerate is None:
+                delta = dt.datetime.fromisoformat(
+                    frames[1].get("T")
+                ) - dt.datetime.fromisoformat(frames[0].get("T"))
+                framerate = int(round(1 / delta.total_seconds()))
+
+    return periods, framerate
+
+
+def create_links_from_mat_info(
+    filepath_mat_info: Union[str, Path]
+) -> Tuple[Dict[str, Dict[int, int]], Dict[str, Dict[str, int]]]:
+    """Parses the DFL Match Information XML file for unique jIDs (jerseynumbers) and
+    creates two dictionaries, one linking pIDs to jIDs and one linking jIDs to xIDs in
+    ascending order.
+
+    Parameters
+    ----------
+    filepath_mat_info: str or pathlib.Path
+        Full path to XML File where the Match Information data in DFL format is saved
+
+    Returns
+    -------
+    links: Dict[str, Dict[int, int]]
+        A link dictionary of the form `links[team][jID] = xID`.
+    id_to_jrsy: Dict[str, Dict[str, int]]
```

Comment:
This variable name is a bit confusing, maybe change them to `links_jID_to_xID` and `links_pID_to_jID`, respectively?